src/qml/editorwidgets/CheckBox.qml

```diff
@@ -12,17 +12,18 @@ Item {
 
   CheckBox {
     property var currentValue: value
+    property bool isBool: value === true || value === false
```

Comment:
```suggestion
    readonly property bool isBool: field.type === 1
```

Reply:
And a comment (something like "If the field type is boolean, ignore the configured 'CheckedState' and 'UncheckedState' values and work with true/false always")

Reply:
That's not working:

![image](https://user-images.githubusercontent.com/28384354/54983083-68a31c00-4fac-11e9-9b92-f71c9aaef515.png)

Reply:
But yes - that's the part I don't like neither. 

But in the end it's a hack that is understandable enough that we can keep it IMO.

Reply:
`field.type == 1`?

Reply:
... I think we need this approach or if the default value is NULL it will fail to detect the type.

Reply:
that works

Reply:
great, with Qt 5.13 apparently enum support will be improved, whatever that means, let's keep it in mind